### ISSUE TITLE
User has to re-enter license key after every restart (Paddle 3.0.32)

### ISSUE THREAD

**Issue by tunabelly** [PaddleHQ/Mac-Framework#194]
I have a customer that's having to re-enter his license key for TG Pro every time he restarts his Mac. I'm surprised this is still happening since we've gone though at least 4 revisions of the framework where this would occur.

He's using TG Pro 2.29 which includes Paddle 3.0.32 so it's pretty current. I've also emailed the padl file to @GregularExpressions so hopefully that helps.

**Comment by yangyubo** [comment]
Same issue for one of our customers, hope this will be fixed soon.

PS: the app is sandboxed.

**Comment by GregularExpressions** [comment]
Hey @yangyubo 

Sorry to hear you're having a similar issue, I'm working with Tunabelly over email directly to resolve theirs.

If you wouldn't mind, could you email me (greg at paddle dot com) more details about this user - their email and license key would be helpful.

Thanks!
Greg.

**Comment by yangyubo** [comment]
@GregularExpressions  the customer has reported that the license window not show again after re-activiate and restart. So I think this might be a different issue, I will collect more cases and get back to you if necessary.

**Comment by pixeleyesltd** [comment]
We're experiencing a similar issue. Any update on this issue? Cheers!

**Comment by GregularExpressions** [comment]
Hey @pixeleyesltd, these tend to be relatively isolated cases/situations - could you elaborate further please? What sort of behaviour is the end user seeing?

Thanks,
Greg.

**Comment by pixeleyesltd** [comment]
Any update?

**Comment by GregularExpressions** [comment]
Hey @pixeleyesltd if you could arrange to email me greg at paddle dot com a copy of the end users' paddle licensing files that would be really helpful.

They'll potentially be in both of these places and we'd like both if they are:

`~/Library/Application Support/<App Name>`
`/Users/Shared/<App Name>`

Thanks!
Greg.

**Comment by tunabelly** [comment]
@GregularExpressions this is still happening with Paddle 3.0.34. We haven't released an update yet with 3.0.37. Will that resolve it? 

I have another angry customer, in his words:

> I started with a trial of your product and later purchased.  I have entered the license key at least 6 times, but every time I shutdown, the next startup it says trial over and I am forced to enter the license key.  What is the solution???

**Comment by tunabelly** [comment]
@GregularExpressions I also just cc'ed you in the reply to the customer to see if there's a workaround or something you can do to help him.

**Comment by tunabelly** [comment]
@GregularExpressions the customer is getting really upset that he has to do this EVERY SINGLE TIME he restarts his Mac. In his words:

> This is getting old!  There has got to be a way to fix this!

This HAS to be fixed in the next Paddle update.

**Comment by developedsoftware** [comment]
Just to echo tunabelly. Surely an issue like this should not still be occuring a year after it was first reported? Any response from paddle?

**Comment by tunabelly** [comment]
@developedsoftware I'm actually having a lot of paying customers complaining about this over the last few weeks. I've even had to refund money now because it's so frustrating to them.

I've contacted numerous people at Paddle (since customers are getting very upset) and they are now looking into it with some senior devs/support people. To answer your questions, yes it's still occurring, and now after quite a few emails Paddle is working on it.

**Comment by tunabelly** [comment]
@GregularExpressions we still don't have an answer for our customers. I've emailed other contacts at Paddle to help get this sorted out and they have been helpful although I'm getting really worried (and customers are getting very anxious) about a proper solution not appearing. 

It sounds like some type of permissions issue where the Paddle SDK can't write the license file out properly?

**Comment by lukeenglish** [comment]
I know this is probably easier said than done. But perhaps it’s time to ditch the framework and find one that you get some support from. I know I wouldn’t be using it in my software! 

Thanks for keeping us updated tunabelly! Much appreciated

**Comment by tunabelly** [comment]
@lukeenglish no problem. It's a huge pain since I can't just jump to V4 since that will inevitably come with its own set of issues. 

I'm giving up on a fix from Paddle for this.

**Comment by rishimodha** [comment]
@tunabelly We finally got one user write in to us about the exact same issue. We've never used any SDK other than v3 so it's not related to the older v2 SDK. This was a recent customer purchase of our app and the app is losing it's activation after every restart.

My feeling right now is that it must be a user config issue as we cannot reproduce the issue and never have had this reported to us before.

@tunabelly did you ever make any further progress on this and do you have a workaround for your customers?

**Comment by rishimodha** [comment]
@tunabelly @GregularExpressions  Happy to report we have resolved this issue for our customer.

See steps below - hopefully it helps someone else too:

1. Download an Uninstaller app that fully removes ALL traces of the app in question. We got our customer to use our Delete Apps: Uninstaller app which is available on the Mac App Store for free.
https://apps.apple.com/us/app/delete-apps-uninstaller/id1033808943?mt=12

2. Use Delete Apps to remove all app traces

3. Download and install fresh copy of the app the customer has purchased

4. License the app

5. Enjoy the app without any further problems :)

**Comment by tunabelly** [comment]
@rishimodha that sound like a similar method a customer used to try and resolve it which seemed to work. He removed the license key from ~/Users/Shared/TG Pro/ and re-activated the app and it kept the license key properly.

**Comment by tunabelly** [comment]
@rishimodha Just as a side note, I've been in contact with the seller team offline and I provided all the information necessary of how we use the Paddle SDK and many, many customers have also sent the PADL files to them for inspection.

Unfortunately they can't replicate it and have no idea how to fix it.

**Comment by rishimodha** [comment]
@tunabelly @GregularExpressions We are getting more customers with this same issue writing in. Do you really have no idea on how to fix it?

**Comment by tunabelly** [comment]
@GregularExpressions @JoMarques09 we have more customers writing in with this issue. 

What do we tell them other than "It's a bug in the Paddle service we use and they have no idea how to fix it"? 

We can't send them to Paddle support because that leads to customers getting more frustrated from a circular pattern with no answer.

We suggest they remove license key from ~/Users/Shared/TG Pro/ and re-activate the app. This isn't really a fix but a clunky workaround.

**Comment by tunabelly** [comment]
@GregularExpressions @JoMarques09 we're getting a large number of customers writing in about this and they are super frustrated! 

How do we fix this for them????

**Comment by JoMarques09** [comment]
Hi @tunabelly,

This issue is now next on or list of priorities, but in the meantime, I would recommend either remove license key from ~/Users/Shared/TG Pro/ and re-activate the app as you suggested or use an uninstall app as @rishimodha described above.

I've also had a word with the buyer support team and instruct them to recommend the workarounds described above whenever they get a user complaining about this issue.

**Comment by tunabelly** [comment]
@JoMarques09 thanks for letting the support team know to tell customers about the workaround for now. Since this issue has been in progress for the last 6 days, when can we expect a fix? I'd like to get this out sooner than later since we'll have to do another update to the app.

**Comment by tunabelly** [comment]
@JoMarques09 what's the update on this? 

It's almost impossible to plan our next update without knowing when this fix will be available.

**Comment by tunabelly** [comment]
@JoMarques09 can we please get an update on this?

**Comment by tunabelly** [comment]
@JoMarques09 @GregularExpressions what the heck is going on? Is anyone at Paddle reading these anymore?

**Comment by tunabelly** [comment]
I've been trying to get an answer here for weeks, and I've been emailing seller support and now I'm trying my other contact at Paddle.

**Silence, that's all I'm getting.**

It's like Paddle has completely forgotten about us Mac developers...

**Comment by JoMarques09** [comment]
Hi @tunabelly my apologies I completely missed your messages here as I did not get any notifications, but I can see that Vikki has been in contact with you.

Unfortunately, priorities shifted a bit in the past weeks. I'll talk to the team today to see where we are at with regards to this issue and get back to you.

**Comment by tunabelly** [comment]
@JoMarques09 yes you're correct, Vikki got in touch with my October 28. I appreciate you getting back to me and sending an offline email as well. 

I can't begin to tell you how frustrating this is that I literally just had a customer email about this issue to Paddle support. They then forwarded the email to us asking us to help the customer!!

I understand priorities change, I'm just really disappointed this will never get fixed considering I've been trying to get this resolved for almost 2 years.

**Comment by tunabelly** [comment]
I won't put another message here. Just wanted to clarify that I now have multiple customers that can't resolve this issue with the workarounds. So now I have pissed off paying customers with a very frustrating bug that I can't fix unless I do a mass upgrade from V3 -> V4 (we're talking 10s of thousands of people) for the app.

**Comment by tunabelly** [comment]
Just to clarify for anyone reading, **Paddle will not fix this** (even though the bug is almost 2 years old). So anyone else having this issue like @rishimodha, the only way to resolve this is to upgrade to V4 or use another solution.
